src/pages/auth/register.tsx

```diff
@@ -0,0 +1,65 @@
+import { GetServerSideProps, NextPage } from "next";
+import Link from "next/link";
+import { useRouter } from "next/router";
+
+import { getServerAuthSession } from "@server/common/getServerAuthSession";
+
+const Register: NextPage = () => {
+  const router = useRouter();
+  return (
+    <div>
+      <form method="post" action="/api/user/register">
+        <input
+          name="callbackUrl"
+          type="hidden"
+          defaultValue={router.query.callbackUrl ?? "/"}
+        />
+        <label>
+          Email
+          <input name="email" type="email" />
+        </label>
+        <label>
+          Password
+          <input name="password" type="password" />
+        </label>
+        <label>
+          First Name
+          <input name="first_name" type="text" />
+        </label>
+        <label>
+          Last Name
+          <input name="last_name" type="text" />
+        </label>
+        <button type="submit">Register</button>
+      </form>
+      <Link
+        href={{
+          pathname: "/auth/login",
+          query: { callbackUrl: router.query.callbackUrl },
+        }}
+      >
+        <button>Already Have An Account?</button>
+      </Link>
+    </div>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
```

Comment:
Do we want to define a type for context?

Reply:
https://github.com/arkinmodi/project-sayyara/pull/128#discussion_r997624011